designer/server/src/main/scala/pl/touk/nussknacker/ui/api/NuDesignerErrorToHttp.scala

```diff
@@ -32,21 +34,42 @@ object NuDesignerErrorToHttp extends LazyLogging with FailFastCirceSupport {
 
   def toResponseEither[T: Encoder](either: Either[NuDesignerError, T]): ToResponseMarshallable = either match {
     case Right(t)  => t
-    case Left(err) => nuDesignerErrorToHttp(err)
+    case Left(err) => httpResponseFrom(err)
   }
 
-  def nuDesignerErrorToHttp(error: NuDesignerError): HttpResponse = {
-    val statusCode = error match {
-      case _: NotFoundError         => StatusCodes.NotFound
-      case _: FatalError            => StatusCodes.InternalServerError
-      case _: BadRequestError       => StatusCodes.BadRequest
-      case _: IllegalOperationError => StatusCodes.Conflict
-      // unknown?
-      case _ =>
-        logger.error(s"Unknown NuDesignerError: ${error.getMessage}. Http status 500 will be returned.", error)
-        StatusCodes.InternalServerError
-    }
-    HttpResponse(status = statusCode, entity = error.getMessage)
+  def httpResponseFrom(error: NuDesignerError): HttpResponse = {
+    HttpResponse(
+      status = httpStatusCodeFrom(error),
+      entity = error.getMessage
+    )
+  }
+
+  private def logError(error: NuDesignerError): Unit = error match {
+    case error: NotFoundError =>
+      logger.debug(s"Not found error: ${error.getMessage}. ${returnedHttpStatusInfo(error)}", error)
+    case error: FatalError =>
+      logger.error(s"Fatal error: ${error.getMessage}. ${returnedHttpStatusInfo(error)}", error)
+    case error: BadRequestError =>
+      logger.debug(s"Bad request error: ${error.getMessage}. ${returnedHttpStatusInfo(error)}", error)
+    case error: IllegalOperationError =>
+      logger.error(
```

Comment:
This "error" appears e.g. when smb invokes an action on a scenario and the scenario is in a state that doesn't allow this operation.   It is a kind of problem on the client side. We don't have control on the client side, so the client can generate a flood of error logs. I recommend using debug in this place instead.

Reply:
After discussion, I've decided to modify FatalError and OtherError too, to be consistent with the described above strategy.